core/src/main/scala/kafka/common/ClientIdAndBroker.scala

```diff
@@ -26,6 +26,10 @@ trait ClientIdBroker {
 }
 
 case class ClientIdAndBroker(clientId: String, brokerHost: String, brokerPort: Int) extends ClientIdBroker {
+
+  // ipv6 addresses contain ':' which is not a valid character in metrics ObjectName
+  def brokerHostMetricsTag = brokerHost.replace(':', '.')
```

Comment:
@granthenke Thank you for the review. I was thinking earlier that perhaps we dont want to change a tag that contained colon by mistake. But I suppose it is better to change any occurrence of colon, making the metric name valid. I have moved the code to `KafkaMetricsGroup`.